modules/ROOT/pages/secure-storage.adoc

```diff
@@ -0,0 +1,110 @@
+
+
+== Update the Master Password
+
+You can change or rotate the master password for a particular provider. When changing a master password, all the encrypted data by this provider is re-encrypted with the new master password.
+
+=== Before you Begin
+
+* Ensure that all other instances of Anypoint Studio and any other Eclipse-based application are not running. +
+Because the secure storage file is shared across all running instances and the master password is cached in each instance's memory, updating your master password causes other opened instances to still write the credentials using the old master password. Closing all other instances ensures you avoid decryption issues. See <<errors, Decryption Errors>> for more information.
+
+To re-generate your master password:
+
+. Navigate to *Preferences* > *General* > *Security* > *Secure Storage*.
+. Select *Change Password...*:
++
+image::secure-storage-change-password.png[]
+. Studio prompts you to create a password hint for this new password. +
+Select *Yes* to generate a new password hint:
++
+image::new-password-created.png[]
++
+[CAUTION]
+--
+If you don't supply new password hints, then the former ones persist, which reference the former master password. You cannot recover this new master password using the previous password hints.
+--
+. Studio prompts you that the password has been successfully updated.
+. Select *Apply and Close*.
+
+
+== Customize the Location of the Secure Storage and Password Files
+
+To change the location of the password and secure storage files:
+
+. Locate your `AnypointStudio.ini` file:
+* In Windows:
+.. Go to your Anypoint Studio's installation directory.
+* In macOS:
+.. Go to your `/Applications` folder, right-click the `AnypointStudio` package and select *Show Package Contents*.
+.. Go to `Contents` > `Eclipse`, and open the `AnypointStudio.ini` file.
+* In Linux:
+.. Go to `/usr/lib/anypointstudio/anypointstudio.ini`.
+. Add the following options together to bypass the password provider mechanism:
++
+[source]
+--
+-eclipse.keyring <file_path> //<1>
+-eclipse.password <file_path> //<2>
+--
+<1> Configures the location of the Secure Storage file.
+<2> Configures the location of the password file. Studio uses this file's content as the master password.
+
+[CAUTION]
+You can only decrypt your credentials with the same password file used to encrypt them.
+
+[[errors]]
+== Decryption Errors
+
+=== No Password Provided
+
+Exception: `org.eclipse.equinox.security.storage.StorageException: No password provided`
+
+This error occurs when Studio loads and caches the master password for the provider and then it tries to decrypt a value that was encrypted with another master password.
+
+This happens when a user changes the master password using one instance of Studio or another Eclipse-based application changes the master password while another instance of Studio saves a new credential using a cached master password.
```

Comment:
Do we need both the texts at Line 99 and Line 101: are they documenting the same reason, just using different words?